core/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/JpaEventStorageEngine.java

```diff
@@ -391,6 +392,27 @@ protected void storeSnapshot(DomainEventMessage<?> snapshot, Serializer serializ
         return Optional.ofNullable(results.get(0));
     }
 
+    @Override
+    public TrackingToken createHeadToken() {
+        List<Long> results = entityManager().createQuery("SELECT MAX(e.globalIndex) FROM " + domainEventEntryEntityName() + " e", Long.class)
+                                          .getResultList();
+        if (results.size() == 0 || results.get(0) == null) {
+            return null;
+        }
+        return GapAwareTrackingToken.newInstance(results.get(0), Collections.emptySet());
+    }
+
+    @Override
+    public TrackingToken createTokenAt(Instant dateTime) {
+        List<Long> results = entityManager().createQuery("SELECT MAX(e.globalIndex) FROM " + domainEventEntryEntityName() + " e WHERE e.timeStamp < :dateTime", Long.class)
```

Comment:
Same as JDBC implementation, select the smallest globalIndex for an event on or after given date. Subtract one from the result to ensure first event is included in the `openStream`